zerver/lib/retention.py

```diff
@@ -171,30 +171,29 @@ def move_messages_to_archive(message_ids: List[int]) -> None:
     messages = list(Message.objects.filter(id__in=message_ids).values())
     if not messages:
         raise Message.DoesNotExist
-    arc_messages = []
-    for message in messages:
-        arc_message = ArchivedMessage(**message)
-        arc_messages.append(arc_message)
-    ArchivedMessage.objects.bulk_create(arc_messages)
+
+    ArchivedMessage.objects.bulk_create([ArchivedMessage(**message) for message in messages])
+
     # Move user_messages to the archive.
     user_messages = UserMessage.objects.filter(
-        message_id__in=message_ids).exclude(id__in=ArchivedUserMessage.objects.all())
-    archiving_messages = []
-    for user_message in user_messages.values():
-        archiving_messages.append(ArchivedUserMessage(**user_message))
-    ArchivedUserMessage.objects.bulk_create(archiving_messages)
+        message_id__in=message_ids).exclude(id__in=ArchivedUserMessage.objects.all()).values()
+    user_messages_ids = [user_message['id'] for user_message in user_messages]
+    ArchivedUserMessage.objects.bulk_create(
```

Comment:
I could be misremembering, but I think `bulk_create` might return the IDs it created?